marimo/_plugins/ui/_impl/array.py

```diff
@@ -96,7 +96,10 @@ def elements(self) -> Sequence[UIElement[JSONType, object]]:
     def _convert_value(self, value: dict[str, JSONType]) -> Sequence[object]:
         if self._initialized:
             for k, v in value.items():
-                self._elements[int(k)]._update(v)
+                element = self._elements[int(k)]
+                # only call update if the value has changed
+                if element and element._value != v:
+                    element._update(v)
```

Comment:
@akshayka, one of the bugs that was fixed (sending only partial of the dictionary plugin) surfaced another bug which was calling `._update` on all elements of `mo.ui.array` (or `mo.ui.dict`) which would call the `on_change` handler. i noticed this with this code which would call `on_click` when it wasnt really clicked

```python
    dict = mo.ui.dictionary(
        {
            "slider": mo.ui.slider(1, 10),
            "text": mo.ui.text("type something!"),
            "array": mo.ui.array(
                [
                    mo.ui.button(value=0, on_click=lambda v: v + 1)
                    for _ in range(3)
                ],
                label="buttons",
            ),
        }
    )
```